msmarco-passage-elser-throughput/track.py

```diff
@@ -0,0 +1,141 @@
+#     try:
+#         await es.ml.put_trained_model(model_id=elser_model_id, input={"field_names": "text_field"})
+#         return True
+#     except BadRequestError as bre:
+#         if (
+#             bre.body["error"]["root_cause"][0]["reason"]
+#             == "Cannot create model [.elser_model_1] the id is the same as an current model deployment"
+#             or bre.body["error"]["root_cause"][0]["reason"]
+#             == "Trained machine learning model [.elser_model_1] already exists"
+#         ):
+#             return True
+#         else:
+#             print(bre)
+#             return False
+#     except Exception as e:
+#         print(e)
+#         return False
+
+
+async def put_elser(es):
+    try:
+        await es.perform_request(method="PUT", path="/_ml/trained_models/.elser_model_1", body={"input": {"field_names": ["text_field"]}})
+    except BadRequestError as bre:
+        if (
+            bre.body["error"]["root_cause"][0]["reason"]
+            == "Cannot create model [.elser_model_1] the id is the same as an current model deployment"
+            or bre.body["error"]["root_cause"][0]["reason"] == "Trained machine learning model [.elser_model_1] already exists"
+        ):
+            return True
+        else:
+            print(bre)
+            return False
+    except Exception as e:
+        print(e)
+        return False
+
+
+async def poll_for_elser_completion(es):
+    try_count = 0
+    max_wait_time_seconds = 120
+    wait_time_per_cycle_seconds = 5
+    while wait_time_per_cycle_seconds * try_count < max_wait_time_seconds:
+        try:
+            response = await es.ml.get_trained_models(model_id=elser_model_id, include="definition_status")
+            if is_model_fully_defined(response):
+                return True
+        except NotFoundError:
+            print("\nwaiting... try count:", try_count, end="")
+            await asyncio.sleep(wait_time_per_cycle_seconds)
+            try_count += 1
+    print()
+    return False
+
+
+def is_model_fully_defined(response):
+    return response["trained_model_configs"][0]["fully_defined"]
+
+
+async def stop_trained_model_deployment(es):
+    try:
+        await es.ml.stop_trained_model_deployment(model_id=elser_model_id, force=True)
+        return True
+    except BadRequestError as bre:
+        if model_deployment_already_exists(bre):
+            return True
+        else:
+            print(bre)
+            return False
+
+
+async def start_trained_model_deployment(es, params):
+    number_of_allocations = params["number_of_allocations"]
+    threads_per_allocation = params["threads_per_allocation"]
+    queue_capacity = params["queue_capacity"]
+    try:
+        await es.ml.start_trained_model_deployment(
+            model_id=elser_model_id,
+            wait_for="fully_allocated",
+            number_of_allocations=number_of_allocations,
+            threads_per_allocation=threads_per_allocation,
+            queue_capacity=queue_capacity,
```

Comment:
Model deployments have cache for inference results, if that cache is being hit then that is faster than inference. Can you please disable the cache just to be sure it is not affecting the results.

`cache_size='0b'`

https://www.elastic.co/guide/en/elasticsearch/reference/current/start-trained-model-deployment.html#start-trained-model-deployment-query-params